sdk/typescript/runtime/main.go

```diff
@@ -98,28 +103,37 @@ func (t *TypescriptSdk) Codegen(
 		return nil, fmt.Errorf("failed to analyze module config: %w", err)
 	}
 
-	runtimeBaseCtr := runtimeBaseContainer(cfg, t.SDKSourceDir).
-		withConfiguredRuntimeEnvironment().
-		withGeneratedSDK(introspectionJSON).
-		withSetupPackageManager().
-		withGeneratedLockFile().
-		withUserSourceCode()
+	var codegen *dagger.Directory
+	switch cfg.runtime {
+	case Bun:
+		codegen, err = NewBunRuntime(cfg, t.SDKSourceDir, introspectionJSON).GenerateDir(ctx)
+	case Deno:
+		codegen, err = NewDenoRuntime(cfg, t.SDKSourceDir, introspectionJSON).GenerateDir(ctx)
+	case Node:
+		codegen, err = NewNodeRuntime(cfg, t.SDKSourceDir, introspectionJSON).GenerateDir(ctx)
+	default:
+		return nil, fmt.Errorf("unknown runtime %s", cfg.runtime)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate runtime dir: %w", err)
+	}
 
-	if runtimeBaseCtr, err = runtimeBaseCtr.addInitTemplateIfNoUserFile(ctx); err != nil {
-		return nil, err
+	// TODO: handle that in an init method.
+	// Add default template if no source files exist
+	sourceFiles, err := cfg.source.Glob(ctx, "src/**/*.ts")
+	if err != nil {
+		return nil, fmt.Errorf("failed to list source files: %w", err)
 	}
 
-	// Extract codegen directory
-	codegen := dag.
-		Directory().
-		WithDirectory(
-			"/",
-			runtimeBaseCtr.ModuleDirectory(),
-			dagger.DirectoryWithDirectoryOpts{Exclude: []string{"**/node_modules", "**/.pnpm-store"}},
-		)
+	if len(sourceFiles) == 0 {
+		codegen = codegen.WithFile(
+			"src/index.ts",
+			dag.File("index.ts", tsutils.TemplateIndexTS(strcase.ToCamel(cfg.name))))
+	}
```

Comment:
```suggestion
		codegen = codegen.WithNewFile(
			"src/index.ts",
			tsutils.TemplateIndexTS(strcase.ToCamel(cfg.name)))
	}
```

Reply:
Done in [6e88a0d](https://github.com/dagger/dagger/pull/11309/commits/6e88a0d3a265e0a38940b6bf8dea474f4bd02727)